lib/cext/preprocess.rb

```diff
@@ -305,6 +305,66 @@ def read_write_array(name, leaking)
         replacement: BYEBUG_THREADS_INIT
       }
     ]
+  },
+  'parser.c' => {
+    gem: 'json',
+    ext_dir: 'parser',
+    patches: [
+      *read_write_field('json','Vsource', false),
+      *read_write_field('json','create_id', false),
+      *read_write_field('json','object_class', false),
+      *read_write_field('json','array_class', false),
+      *read_write_field('json','decimal_class', false),
+      *read_write_field('json','match_string', false),
+      { # cParser_initialize
+        match: /if \(rb_tr_managed_from_handle\(json->Vsource\)\)/,
+        replacement: 'if (rb_tr_managed_from_handle_or_null(json->Vsource))'
+      },
```

Comment:
And other locations do not want to handle `null`?
It seems harmless, I'm actually thinking `rb_tr_managed_from_handle` should always check for `null`.

Reply:
The other locations can't be `NULL` because that's a native value and they're expecting a managed object. You end up with errors like:

`Cannot resolve native handle: 0x0 UnsupportedOperationException com.oracle.truffle.llvm.runtime.LLVMContext.getManagedObjectForHandle(LLVMContext.java:333)`

And we can't use `nil` as a simple predicate check. Another way to patch this is to use `NIL_P`.

Reply:
Why are we matching against something that already uses our API? I thought these patches applied to the original source?

Reply:
The original source is `if (json->Vsource)`, which is fine when it's a native pointer (and it its because it's a struct field). But the earlier preprocess step `*read_write_field('json','Vsource', false)` rewrites cases it shouldn't. So, I either need to unwind that in this case or not use the `read_write_field` helper.